src/jsp/jsp_cl.cpp

```diff
@@ -1322,34 +1401,45 @@ jsp_make_method_sig_list (PARSER_CONTEXT *parser, PT_NODE *node, method_sig_list
 	    goto end;
 	  }
 
-	if (sig_num_args > 0)
+	if (param_cnt > 0)
 	  {
-	    sig->arg_info->arg_mode = (int *) db_private_alloc (NULL, (sig_num_args) * sizeof (int));
+	    sig->arg_info->arg_mode = (int *) db_private_alloc (NULL, (param_cnt) * sizeof (int));
 	    if (!sig->arg_info->arg_mode)
 	      {
 		sig->arg_info->arg_mode = nullptr;
 		error = ER_OUT_OF_VIRTUAL_MEMORY;
 		goto end;
 	      }
 
-	    sig->arg_info->arg_type = (int *) db_private_alloc (NULL, (sig_num_args) * sizeof (int));
+	    sig->arg_info->arg_type = (int *) db_private_alloc (NULL, (param_cnt) * sizeof (int));
 	    if (!sig->arg_info->arg_type)
 	      {
 		sig->arg_info->arg_type = nullptr;
 		error = ER_OUT_OF_VIRTUAL_MEMORY;
 		goto end;
 	      }
 
-	    for (int i = 0; i < sig_num_args; i++)
+	    sig->arg_info->default_value = (char **) db_private_alloc (NULL, (param_cnt) * sizeof (char *));
+	    sig->arg_info->default_value_size = (int *) db_private_alloc (NULL, (param_cnt) * sizeof (int));
+
+	    for (int i = 0; i < param_cnt; i++)
 	      {
 		sig->arg_info->arg_mode[i] = sig_arg_mode[i];
 		sig->arg_info->arg_type[i] = sig_arg_type[i];
+
+		sig->arg_info->default_value_size[i] = sig_arg_default_size[i];
+		if (sig->arg_info->default_value_size[i] > 0)
+		  {
+		    sig->arg_info->default_value[i] = sig_arg_default[i];
+		  }
```

Comment:
no need "else clause"?
not defined sig->arg_info->default_value[i] in case of "sig->arg_info->default_value_size[i] <= 0" ?

Reply:
else is not required. `default_value_size > 0` means the `default_value` is packed together.